Author: Apollon Oikonomopoulos <apoikos@debian.org>
Description: Use the system LibYAML instance
 Do not statically link against LibYAML, use dynamic linking against the system
 instance instead.
Last-Update: 2017-08-29
Forwarded: no
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,6 +121,9 @@
         INCLUDE_DIRECTORIES(${LIBUV_INCLUDE_DIRS})
         LINK_DIRECTORIES(${LIBUV_LIBRARY_DIRS})
     ENDIF (LIBUV_FOUND)
+    PKG_CHECK_MODULES(LIBYAML REQUIRED yaml-0.1)
+    INCLUDE_DIRECTORIES(${LIBYAML_INCLUDE_DIRS})
+    LIST(INSERT EXTRA_LIBS 0 ${LIBYAML_LIBRARIES})
 ENDIF (PKG_CONFIG_FOUND)
 IF (NOT LIBUV_FOUND)
     FIND_PACKAGE(LibUV)
@@ -313,7 +316,6 @@
 
 SET(UNIT_TEST_SOURCE_FILES
     ${LIB_SOURCE_FILES}
-    ${LIBYAML_SOURCE_FILES}
     ${BROTLI_SOURCE_FILES}
     deps/picotest/picotest.c
     t/00unit/test.c
@@ -445,7 +447,6 @@
 # standalone server directly links to libh2o using evloop
 SET(STANDALONE_SOURCE_FILES
     ${LIB_SOURCE_FILES}
-    ${LIBYAML_SOURCE_FILES}
     ${BROTLI_SOURCE_FILES}
     deps/neverbleed/neverbleed.c
     src/main.c
